### PR TITLE
Add control server flags for watchdog options

### PR DESCRIPTION
**Pull request by RebeccaMahany**
Relates to https://github.com/kolide/launcher/issues/1224

Gets some boilerplate out of the way for setting `disable_watchdog`, `watchdog_memory_limit`, `watchdog_utilization_limit`, and `watchdog_delay` osquery flags via control server.

### CODE REVIEW

File: ee/agent/flags/flag_controller.go [hunk 1]
```diff
@@ -356,6 +356,46 @@ func (fc *FlagController) OsqueryVerbose() bool {
...
 
+func (fc *FlagController) SetEnableWatchdog(enable bool) error {
+	return fc.setControlServerValue(keys.EnableWatchdog, boolToBytes(enable))
+}
+func (fc *FlagController) EnableWatchdog() bool {
+	return NewBoolFlagValue(WithDefaultBool(fc.cmdLineOpts.EnableWatchdog)).get(fc.getControlServerValue(keys.EnableWatchdog))
+}
+
+func (fc *FlagController) SetWatchdogDelaySec(sec int) error {
+	return fc.setControlServerValue(keys.WatchdogDelaySec, intToBytes(sec))
```
Comment (James-Pickett) [inline ee/agent/flags/flag_controller.go:367]: Is the need to use an int here driven by the osquery options? What I'm getting at is I wonder if we need to introduce another value type to flags or can we make due with existing float and duration.

Comment (RebeccaMahany) [reply ee/agent/flags/flag_controller.go:367]: For the delay sec, we could do a duration and convert it to seconds when setting the osquery flag -- but I don't think we should let memory or CPU limit be a float when osquery expects both of those to be integers. It wasn't burdensome to add/maintain the int type and I figure if we don't need it now, we'd want it at some point in the future anyway.

**Review by James-Pickett — APPROVED** [review]

File: ee/agent/flags/flag_value_int_test.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+			expectedValue:      100,
+		},
+		{
+			name:               "control server with default, min -- does not need clamp",
+			options:            []intOption{WithIntValueDefault(3), WithIntValueMin(0)},
+			controlServerValue: []byte("1"),
+			expectedValue:      1,
+		},
+		{
+			name:               "control server with default, max -- does not need clamp",
```
Comment (zackattack01) [inline ee/agent/flags/flag_value_int_test.go:76]: these look pretty exhaustive, great tests 🔥

**Review by zackattack01 — APPROVED** [review]
nice!

File: pkg/launcher/options_test.go [hunk 3]
```diff
@@ -242,28 +242,32 @@ func getArgsAndResponse() (map[string]string, *Options) {
...
+		OsquerydPath:                    windowsAddExe("/dev/null"),
+		OsqueryHealthcheckStartupDelay:  10 * time.Minute,
+		Transport:                       "grpc",
+		UpdateChannel:                   "stable",
+		AutoloadedExtensions:            []string{"some-extension.ext"},
+		DelayStart:                      0 * time.Second,
+		EnableWatchdog:                  false,
+		WatchdogDelaySec:                60,
+		WatchdogMemoryLimitMB:           200,
+		WatchdogUtilizationLimitPercent: 10,
```
[the remark this answers is not in the document]
Comment (RebeccaMahany) [reply pkg/launcher/options_test.go:270]: Sure thing, I will update in https://github.com/kolide/launcher/pull/1506!

File: ee/agent/flags/flag_controller.go [hunk 4]
```diff
@@ -356,6 +356,46 @@ func (fc *FlagController) OsqueryVerbose() bool {
 	return NewBoolFlagValue(WithDefaultBool(fc.cmdLineOpts.OsqueryVerbose)).get(fc.getControlServerValue(keys.OsqueryVerbose))
 }
 
+func (fc *FlagController) SetEnableWatchdog(enable bool) error {
+	return fc.setControlServerValue(keys.EnableWatchdog, boolToBytes(enable))
+}
+func (fc *FlagController) EnableWatchdog() bool {
```
[the remark this answers is not in the document]
Comment (RebeccaMahany) [reply ee/agent/flags/flag_controller.go:362]: You're right -- I thought we had precedence for naming it this way, but we do not. I will update in a new PR.

[the remark this answers is not in the document]
Comment (RebeccaMahany) [reply ee/agent/flags/flag_controller.go:362]: https://github.com/kolide/launcher/pull/1508

File: pkg/launcher/options_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (RebeccaMahany) [reply pkg/launcher/options_test.go:270]: Done: https://github.com/kolide/launcher/pull/1506/commits/2070c93017948c79f639759c309aaa734bd96e19

**Pull request merged**